lib/openfoodfacts.dart

```diff
@@ -307,7 +307,20 @@ class OpenFoodAPIClient {
   }) async {
     final Response response = await configuration.getResponse(user, queryType);
     final String jsonStr = _replaceQuotes(response.body);
-    final SearchResult result = SearchResult.fromJson(json.decode(jsonStr));
+    final SearchResult result;
+    try {
+      result = SearchResult.fromJson(json.decode(jsonStr));
+    } on FormatException {
+      if (jsonStr.allMatches('/cgi/reset_password.pl').length == 3) {
```

Comment:
I don't speak regexp and I don't know what is expected in the `String`: you're looking for `'/cgi/reset_password.pl'`, right?
Maybe if you're looking for `'\/cgi\/reset_password.pl'` and don't use your `== 3` test it will be easier to read.